src/Microsoft.ML.Core/EntryPoints/EntryPointModuleAttribute.cs

```diff
@@ -9,13 +9,15 @@ namespace Microsoft.ML.Runtime.EntryPoints
     /// <summary>
     /// This is a signature for classes that are 'holders' of entry points and components.
     /// </summary>
-    public delegate void SignatureEntryPointModule();
+    [BestFriend]
+    internal delegate void SignatureEntryPointModule();
 
     /// <summary>
     /// A simplified assembly attribute for marking EntryPoint modules.
     /// </summary>
     [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
-    public sealed class EntryPointModuleAttribute : LoadableClassAttributeBase
+    [BestFriend]
+    internal sealed class EntryPointModuleAttribute : LoadableClassAttributeBase
```

Comment:
Can we make the `EntryPoint` related methods on `ComponentCatalog` internal as well?

Reply:
Yes, assuredly. Do you feel this PR needs to be longer though? I'm trying to break up the internalization work into multiple phases (hence the numbering, phase 1, phase 2, and so forth). While surely we *can* just keep going forever, I think it would be more helpful if we tried to iterate quickly, rather than insisting we internalize everything in one gigantic PR. So just pointing out "you could have internalized more," while I'm sure well intentioned, is something I already kind of know and understand?

---
In reply to: [234256509](https://github.com/dotnet/machinelearning/pull/1626#discussion_r234256509) [](ancestors = 234256509)